pystan/model.py

```diff
@@ -339,10 +347,13 @@ def __init__(self, file=None, charset='utf-8', model_name="anon_model",
                     '-Wno-unused-function',
                     '-Wno-uninitialized',
                     '-std=c++1y',
-                    "-D_hypot=hypot",
-                    "-pthread",
-                    "-fexceptions",
+                    '-D_hypot=hypot',
+                    '-pthread',
+                    '-fexceptions',
+                    '-DSTAN_THREADS',
+                    '-D_REENTRANT',
```

Comment:
Maybe add a comment about _REENTRANT? Is this a windows requirement?

Reply:
https://stackoverflow.com/questions/2601753/what-is-the-reentrant-flag

I think this is related to TBB (all platforms).

Reply:
@wds15 do you know why we need _REENTRANT? It was somewhere mentioned at some point.

Reply:
https://github.com/stan-dev/math/commit/1ef61d46cc4c46784a24e47691217d559fea28e3